code/modules/surgery/bodyparts/robot_bodyparts.dm

```diff
@@ -161,6 +185,21 @@
 	var/wired = FALSE
 	var/obj/item/stock_parts/cell/cell = null
 
+/obj/item/bodypart/chest/robot/emp_act(severity)
+	. = ..()
+	if(!.) // other emp effects handled in parent
+		return
+	if(owner.incapacitated()) // so it doesn't double up with both legs
+		return
+	to_chat(owner, span_danger("Your [src.name]'s logic boards temporarily become unresponsive!"))
+	if(severity == EMP_HEAVY)
+		owner.Stun(4 SECONDS)
+		owner.Shake(pixelshiftx = 5, pixelshifty = 2, duration = 4 SECONDS)
```

Comment:
I did and it is, I find it a very funny effect that clearly shows what's happening. Standing stuns are often confusing and frustrating because you don't realize that you're actually stunned and think some bug is just preventing you from moving.

 The funny super-shake this causes is excellent at signalling the stun effect - 'jitter' is a stupidly common effect that can arise from almost anything and thus it wouldn't really be informative to the player about the 4-second stun the same way.

Reply:
Eh I checked it out and it looks very goofy but it's probably fine.